warehouse/index-stats/src/main/java/datawave/mapreduce/shardStats/StatsHyperLogMapper.java

```diff
@@ -0,0 +1,274 @@
+    
+    private final Map<String,DataTypeInfo> dataTypeMapping = new HashMap<>();
+    
+    @Override
+    protected void setup(Context context) throws IOException, InterruptedException {
+        if (log.isTraceEnabled()) {
+            log.trace("setup(" + context + ")");
+        }
+        super.setup(context);
+        Configuration conf = context.getConfiguration();
+        
+        // set log level if configured
+        String logLevel = conf.get(STATS_MAPPER_LOG_LEVEL);
+        Level level = Level.toLevel(logLevel, DEFAULT_LOG_LEVEL);
+        log.setLevel(level);
+        log.info("log level set to " + level.toString());
+        
+        // set stats configuration data
+        this.outputTable = new Text(conf.get(OUTPUT_TABLE_NAME));
+        log.info("output table(" + this.outputTable.toString() + ")");
+        
+        this.logOutputInterval = conf.getInt(STATS_MAPPER_OUTPUT_INTERVAL, DEFAULT_OUTPUT_INTERVAL);
+        log.info("output log interval(" + this.logOutputInterval + ")");
+        this.logInputInterval = conf.getInt(STATS_MAPPER_INPUT_INTERVAL, DEFAULT_INPUT_INTERVAL);
+        log.info("input log interval(" + this.logInputInterval + ")");
+        this.sumUniqueCounts = conf.getBoolean(STATS_MAPPER_UNIQUE_COUNT, false);
+        log.info("unique counts(" + this.sumUniqueCounts + ")");
+        
+        // hyperlog precision
+        this.normalPrecision = conf.getInt(HYPERLOG_NORMAL_OPTION, HYPERLOG_NORMAL_DEFAULT_VALUE);
+        log.info("normal precision(" + this.normalPrecision + ")");
+        this.sparsePrecision = conf.getInt(HYPERLOG_SPARSE_OPTION, HYPERLOG_SPARSE_DEFAULT_VALUE);
+        log.info("sparse precision(" + this.sparsePrecision + ")");
+        
+        this.visibility = new ColumnVisibility(conf.get(StatsJob.STATS_VISIBILITY));
+        
+        // initialize type registry
+        TypeRegistry.getInstance(conf);
+        
+        InputSplit split = context.getInputSplit();
+        log.info("Got a split of type " + split.getClass());
+        if (log.isInfoEnabled()) {
+            if (split instanceof TabletSplitSplit) {
+                TabletSplitSplit tabletSplit = (TabletSplitSplit) split;
+                log.info("Has " + tabletSplit.getLength() + " file range splits");
+                for (int i = 0; i < tabletSplit.getLength(); i++) {
+                    FileRangeSplit subSplit = (FileRangeSplit) tabletSplit.get(i);
+                    log.info("Got a file range spit of " + subSplit + " with a range of " + subSplit.getRange());
+                }
+            }
+        }
+        
+        if (log.isTraceEnabled()) {
+            log.trace("Completed setup(" + context + ")");
+        }
+    }
+    
+    @Override
+    protected void map(Key key, Value value, Context context) throws IOException, InterruptedException {
+        if (log.isTraceEnabled()) {
+            log.trace("map(" + key + ", " + value + ")");
+        }
+        
+        if (!setPartitionerContext) {
```

Comment:
Can this block of code be moved to setup?

Reply:
I will have to look into this. This was taken from the original code.

Reply:
Looking at the MultiTableRangePartitioner class, this should be in the setup method.